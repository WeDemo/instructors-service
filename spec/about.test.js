import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import About from '../client/src/components/about.jsx';

const renderer = require('react-test-renderer');
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });



describe('About Component Snapshot', () => {
  const defaultProps = {
    info: {
      instInfo: {
        photo_url: 'google.com',
        rating: 5,
        reviews: 10,
        students: 100,
        courses: 10,
        inst_name: 'Charlie',
        title: 'Teacher',
        blurb: 'this is my blurb'
      }
    }
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< About {...defaultProps} />);
  });

  it('renders About component', () => {  
    const AboutTest = (props) => <About {...defaultProps} {...props} />;
    const AboutTestComponent = renderer.create(<AboutTest />).toJSON();
    expect(AboutTestComponent).toMatchSnapshot();
  });

  it('should render a div', () => {  
    expect(wrapper.find('div')).toBeDefined();     
  });
  
  it('should render an instructor class', () => {
    expect(wrapper.exists('.instructor')).toEqual(true);
  });

  it('should render one instructor photo', () => {
    expect(wrapper.find('.instructorPhoto').length).toEqual(1);
  });

  // it('should render more than one blurb paragraph', () => {
  //   expect(wrapper.find('p').length).toBeGreaterThan(1);
  // });
});





