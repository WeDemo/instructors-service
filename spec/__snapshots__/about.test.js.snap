// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component Snapshot renders About component 1`] = `
<div
  className="instructor"
>
  <div
    className="info_photo"
  >
    <img
      className="instructor_photo"
      src="google.com"
    />
    <table
      className="instructor_info"
    >
      <tbody>
        <tr>
          <td>
            <img
              className="instructor_icon"
              src="https://s3-us-west-1.amazonaws.com/u-demo/blackstar.png"
            />
          </td>
          <td>
            <span
              className="instructor_stat"
            >
              5
            </span>
            <span
              className="instructor_stat_text"
            >
              Instructor Rating
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <img
              className="instructor_icon"
              src="https://s3-us-west-1.amazonaws.com/u-demo/chat.png"
            />
          </td>
          <td>
            <span
              className="instructor_stat"
            >
              10
            </span>
            <span
              className="instructor_stat_text"
            >
              Reviews
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <img
              className="instructor_icon"
              src="https://s3-us-west-1.amazonaws.com/u-demo/user.png"
            />
          </td>
          <td>
            <span
              className="instructor_stat"
            >
              100
            </span>
            <span
              className="instructor_stat_text"
            >
              Students
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <img
              className="instructor_icon"
              src="https://s3-us-west-1.amazonaws.com/u-demo/play.png"
            />
          </td>
          <td>
            <span
              className="instructor_stat"
            >
              10
            </span>
            <span
              className="instructor_stat_text"
            >
              Courses
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="title_blurb"
  >
    <div>
      <div
        className="instructor_name"
      >
        Charlie
      </div>
    </div>
    <div
      className="instructor_title"
    >
      Teacher
    </div>
    <div
      className="instructor_blurb short"
    >
      <p>
        this is my blurb
      </p>
      <div
        className="more_container visible"
      >
        <div
          className="more_blurb"
          onClick={[Function]}
        >
          + See more
        </div>
      </div>
    </div>
  </div>
</div>
`;
